### ISSUE TITLE
running_programs_using_decorators fails

### ISSUE THREAD

**Issue by akihikokuroda** [Qiskit-Extensions/quantum-serverless#1090]
### Steps to reproduce the problem
Install quantum middleware locally (M1 Mac for my case)
Execute 'running_programs_using_decorators' in the Jupyter notebook.

### What is the current behavior?
"Hello, Qiskit!" fails
job.logs() shows
```
Traceback (most recent call last):\n  File "/tmp/ray/session_2023-11-14_10-23-00_770226_8/runtime_resources/working_dir_files/_ray_pkg_6635b2a1eae383f6/entrypoint_a80fb97c.py", line 11, in <module>\n    function = cloudpickle.load(file)\nAttributeError: Can\'t get attribute \'_function_setstate\' on <module \'cloudpickle.cloudpickle\' from \'/home/ray/anaconda3/lib/python3.9/site-packages/cloudpickle/cloudpickle.py\'>\n'
``` 

### What is the expected behavior?
Completes without error

**Comment by akihikokuroda** [comment]
This may unique for M1 mac

The notebook image gets cloudpickle 3.0.0 installed and the Ray node gets cloudpickle 2.2.1 installed.  This causes serialization mismatch and causes the failure.

**Comment by akihikokuroda** [comment]
@IceKhan13 @Tansito @psschwei  Is this only for M1 mac(arm64)?  Can somebody try this with amd64 system? Thanks!

**Comment by akihikokuroda** [comment]
Pin cloudpickle to 2.2.1 fix this but I'm not sure if it's the right fix for this yet.

**Comment by psschwei** [comment]
when you say install locally, are you building from source or `pip install quantum-serverless`?

**Comment by akihikokuroda** [comment]
I built it from the main branch.

**Comment by psschwei** [comment]
and running locally using docker compose?

**Comment by akihikokuroda** [comment]
I installed with helm.

**Comment by psschwei** [comment]
even better (that's what I started trying to install with)

**Comment by akihikokuroda** [comment]
The python package is installed at the image build time and won't change after that (I believe).

**Comment by Tansito** [comment]
I can test it in M2 now 😂

**Comment by psschwei** [comment]
it failed for me, but not for the reason it failed for you:

Failed to pull image "registry.access.redhat.com/ubi8/openssl:8.8-9": rpc error: code = Unknown desc = failed to pull and unpack image "registry.access.redhat.com/ubi8/openssl:8.8-9": failed to resolve reference "registry.access.redhat.com/ubi8/openssl:8.8-9": pulling from host registry.access.redhat.com failed with status code [manifests 8.8-9]: 502 Bad Gateway

that (should be) temporary... let me try again

**Comment by psschwei** [comment]
same error:

Traceback (most recent call last):
  File "/tmp/ray/session_2023-11-14_12-46-08_676930_14/runtime_resources/working_dir_files/_ray_pkg_60b88e2da40ac6db/entrypoint_a8565cd6.py", line 8, in <module>
    function = cloudpickle.load(file)
AttributeError: Can't get attribute '_function_setstate' on <module 'cloudpickle.cloudpickle' from '/home/ray/anaconda3/lib/python3.9/site-packages/cloudpickle/cloudpickle.py'>

**Comment by Tansito** [comment]
My error 😂 :

```bash
 => ERROR [gateway 7/8] RUN chown -R 1000:100 /usr/src/app &&    mkdir /usr/src/app/media && chown 1000:100 /usr/src/app/media               22.5s
 => ERROR [scheduler 7/8] RUN chown -R 1000:100 /usr/src/app &&    mkdir /usr/src/app/media && chown 1000:100 /usr/src/app/media             22.5s
------                                                                                                                                             
 > [gateway 7/8] RUN chown -R 1000:100 /usr/src/app &&    mkdir /usr/src/app/media && chown 1000:100 /usr/src/app/media:
22.52 mkdir: cannot create directory ‘/usr/src/app/media’: File exists
------
------
 > [scheduler 7/8] RUN chown -R 1000:100 /usr/src/app &&    mkdir /usr/src/app/media && chown 1000:100 /usr/src/app/media:
22.44 mkdir: cannot create directory ‘/usr/src/app/media’: File exists
```

**Comment by akihikokuroda** [comment]
@Tansito you have to delete /gateway/media directory before you build the gateway image.

**Comment by Tansito** [comment]
Oh, true! Thanks @akihikokuroda 👍

**Comment by akihikokuroda** [comment]
@psschwei So it's not m1 unique.  We can pin cloudpickle to 2.2.1 for now in requirements.txt in client directory.  @WDYT

**Comment by Tansito** [comment]
I confirm, it happens to me too, yeah.

**Comment by psschwei** [comment]
works for me.
do we need to pin in the notebook requirements file too?

**Comment by akihikokuroda** [comment]
It's only in the client/requrements.txt.  I'll change it.  Thanks for helm @psschwei @Tansito

**Comment by Tansito** [comment]
Thanks to you @akihikokuroda for detecting it 👍
